src/bindings/python/src/pyopenvino/graph/ops/constant.cpp

```diff
@@ -201,33 +155,73 @@ void regclass_graph_op_Constant(py::module m) {
 
     constant.def(
         "get_data",
-        [](ov::op::v0::Constant& self) {
-            auto ov_type = self.get_element_type();
-            auto dtype = Common::type_helpers::get_dtype(ov_type);
-            if (ov_type.bitwidth() < Common::values::min_bitwidth) {
-                return py::array(dtype, self.get_byte_size(), self.get_data_ptr());
+        [](ov::op::v0::Constant& self, py::object& dtype, bool copy) {
+            // Destination type was set:
+            if (!dtype.is(py::none())) {
+                py::dtype dst_dtype;
+                if (dtype.is(py::dtype())) {
+                    dst_dtype = dtype.cast<py::dtype>();
+                }
+                else {
+                    dst_dtype = py::dtype::from_args(dtype);
+                }
+                const auto& ov_type = self.get_element_type();
+                const auto dtype = Common::type_helpers::get_dtype(ov_type);
+                // If dtype is the same as Constant type
+                // casting is NOT required, only check copy flag
+                if (dst_dtype == dtype) {
+                    if (copy) {
+                        return Common::array_helpers::array_from_constant_copy(std::forward<ov::op::v0::Constant>(self));
+                    }
+                    else {
+                        return Common::array_helpers::array_from_constant_view(std::forward<ov::op::v0::Constant>(self));
+                    }
+                }
+                // Otherwise always copy:
+                else {
+                    return Common::array_helpers::array_from_constant_copy(std::forward<ov::op::v0::Constant>(self), dst_dtype);   
+                }
+            }
+            // Copy of data in Constant type:
+            else if (copy) {
+                return Common::array_helpers::array_from_constant_copy(std::forward<ov::op::v0::Constant>(self));
+            }
+            // Shared view of data in Constant type:
+            else {
+                return Common::array_helpers::array_from_constant_view(std::forward<ov::op::v0::Constant>(self));
             }
-            return py::array(dtype, self.get_shape(), _get_strides(self), self.get_data_ptr());
         },
+        py::kw_only(),
+        py::arg("dtype") = py::none(),
+        py::arg("copy") = false,
         R"(
             Access to Constant's data - creates a copy of data.
 
+            If `dtype` parameter is applied, function is casting data to
+            desired dtype.
+            If `dtype` is not specified, it's inherited from Constant itself.
+            If `copy` is enabled, `dtype` and Constant's element type must
+            match to create a view.
+            If `copy` is disabled and `dtype` and Constant's element type
+            do not match -- copy is forced by default.
```

Comment:
And if the types do not match then we always return a view, right?

Reply:
Let's go case by case:
```python
arr = np.array([1.0, 1012.5, 0.0, 0.1, 0.5], dtype=np.float32)
c = ov.runtime.opset13.constant(arr, dtype=ov.Type.f32)

c.get_data(dtype=np.float32, copy=False).flags["OWNDATA"]  # False -- view

c.get_data(copy=False).flags["OWNDATA"]  # False -- view, dtype is deduced from Constant

c.get_data(dtype=np.float32, copy=True).flags["OWNDATA"]   # True -- copy occurs

c.get_data(copy=True).flags["OWNDATA"]   # True -- copy occurs, dtype is deduced from Constant

c.get_data(dtype=np.int16, copy=False).flags["OWNDATA"]  # True -- copy is forced by mismatch of types

c.get_data(dtype=np.int16, copy=True).flags["OWNDATA"]   # True -- copy is forced by mismatch of types
```
Maybe docs are written in confusing way, so they need update? Do you have any guideline how to write them clearly?

Reply:
The examples make sense and reflect my understanding of how this flag should work. "Try to return a view by default, if not possible due to types mismatch or when copy=True, then make a copy", so performance is a priority, but can be overridden.

Reply:
One more example with all arguments with default values: `c.get_data()` returns a view, right? Will it fail when type is not representable in numpy, bf16?

Reply:
Does it make sense to hold this reinterpretation of bf16 as f16? It looks to be the only way how we can still zero-copy bf16 data and pass them to the code that relies on this reinterpretation, right?